---
layout: archive
title: "CV"
permalink: /cv/
author_profile: true
redirect_from:
  - /resume
---

{% include base_path %}

Education
=========

* B.S Biochemestry, University of Granada, 2005.
* M.S. in Bioinformatics for health science, University of Pompeu
  Fabra, 2008.
* Ph.D. in Biomedicine, [Center for Genomic
  Regulation](http://www.crg.es), 2011.

GitHub contribution
===================

Community supported project I actively participate:

* [bcbio-nextgen](http://github.com/chapmanb/bcbio-nextgen)
* [cloudbiolinux](http://github.com/chapmanb/cloudbiolinux)
* [bcbio.rnaseq](https://github.com/roryk/bcbio.rnaseq)
* [bioconda](https://github.com/bioconda/bioconda-recipes)
* [multiqc](https://github.com/ewels/MultiQC)

Open projects I lead:

* [seqcluster](http://github.com/lpantano/seqcluster) [20008100](https://www.ncbi.nlm.nih.gov/pubmed/20008100)
* [seqclusterViz](http://github.com/lpantano/seqclusterViz) [26530722](https://www.ncbi.nlm.nih.gov/pubmed/26530722) and [21976421](https://www.ncbi.nlm.nih.gov/pubmed/21976421)  
* [isomiRs](http://github.com/lpantano/isomiRs)
* [DEGreport](http://github.com/lpantano/DEGreport)
* [mirTOP](https://github.com/miRTop/mirtop)
* [mirGFF3](https://github.com/miRTop/mirGFF3)

Work experience
===============

* Research Scientist at Harvard Chan School Bioinformatic Core,
  2017-to date.
  * Development of tools for NGS data analysis
  * Analyst of high-throughput data
  * Leading Bioinformatic analysis in the fibrosis project collaboration between Harvard and Boehringer Ingelheim
* Msc project supervisor at [UOC](www.uoc.edu), 2018 - to date.

* Research Associate at Harvard Chan School Bioinformatic Core,
  2014–2017.
  * Development of tools for NGS data analysis
  * Analyst of high-throughput data
  * Leading Bioinformatic analysis in the fibrosis project collaboration between Harvard and Boehringer Ingelheim
* Post-doct fellow at
  [IBB](http://ibb.uab.es/ibb/index.php?option=com_wrapper&Itemid=127), 2013–2014.
  * Analyzing de-novo transcripts due to inversion
  * Differential expression due to inversion in HapMap population
  * Involve in inversion database development
* Founder and CTO at ASCIDEA (http://ascidea.com), 2011-2014
  * Leading a group of 3 bioinformaticians
  * Cloud computing architect
  * Development of computational pipelines
* Post-doctoral fellow at [Institut de Medicina Predictiva i
  Personalitzada del Cancer](http://www.imppc.org), 2011–2013.
  * small RNA characterization in human sperm samples
* PhD student at [Center for Genomic
  Regulation](http://www.crg.es), 2006–2011.
  * small RNA characterization in human mental diseases
  * Development of tools for miRNA and small RNA analysis
* Msc student at [Granada University](http://www.ugr.es), 2004–2006.


Talks
=====

  <ul>{% for post in site.talks %}
    {% include archive-single-talk-cv.html %}
  {% endfor %}</ul>

Teaching
========

  <ul>{% for post in site.teaching %}
    {% include archive-single-cv.html %}
  {% endfor %}</ul>

Service and leadership
======================

* Founder of [women in bioinformatics](https://www.meetup.com/Cambridge-woman-developers-in-bioinformatics/)
  meet up at cambridge, 2014 – to date.
* Co-founder of [RSG Spain](http://www.rsgspain.org), the spanish
  student group of ISCB student council, 2010.
* Former member of [ISCB student council](http://www.iscbsc.org),
  2010–2014.

* Peer review, NAR journal, 2017
* Programme committee, BOSC, 2014-to date
* Chari, Mind the Gap Barcelona, 2013
* Master Thesis Supervisor, 2014, Universidad Autonoma de Barcelona

* Programme committee, Student Council Symposium
* ISCB Student Council Symposium Long Beach, 2012 - Travel Fellowship
    chair
* Spain, Portugal and North Africa Student Symposium Barcelona, 2012 -
    Program committee chair
* ISCB Student Council Symposium Vienna, 2012 - Chair
* Spain, Portugal and North Africa Student Symposium Malaga, 2011 -
    Chair
* ISCB Student Council Symposium Boston, 2010 - Volunteer

Skills
======

* LINUX SO
* Cloud computing: Amazon EC2, S3, SQS, IAM
* Control version system: svn and git
* High Performance Computing systems: lsf, slurm, sge
* Basic web Server Development: HTML, DHTML, javascript, ccs3, php and cgi
* Advance database management: mysql, sqlite3 and google datastore

* Programming knowledge:
  * Basic C,C++ environment
  * Intermediate Java environment
  * Advance Perl and Python environment
  * Expert in R and Bioconductor for statistics and visualisation

* Analysis of large-scale data:
  * Analysis of microarrays with R/Rstudio
  * SNPs data mining: samtools, BayesSeq, GATK, freebayes, verdict, mutect.
  * RNA-Seq: tuxedo kit, STAR, salmon, sailfish, kallisto, sleuth, DESeq2
and edgeR/limma.
  * Chip-Seq for histone modifications and transcription factor: macs2,
diffbind.
  * Small RNA-Seq: mirdeep2, seqbuster, seqcluster, targetscan custom

* Biostatistics:
  * Machine learning applications for phenotype classification of
case/control studies
  * Regression model application to data integration


Courses
=======

* Fundamentals and Applications of Microfluidics @ Harvard Extension School 2018
* Tissue Engineering for Clinical Applications @ Harvard Extension School 2017
* Principles of fMRI 1 @ Coursera 2016
* Practical Machine Learning @ Coursera 2016
* Calculus One @ Coursera 2015
* Regression Model @ Coursera 2014
* HPC programming (not finished) @ Coursera 2013
* Analysis of hight-throughput sequencing data @ EMBL-EBI 2012
* Web applications @ Udacity (CS253) 2012
* Machine learning @ Coursera 2012
* Introduction to Artificial Intelligence (www.ai-class.org) 2011
* Advance R (CRG) 2008


Publications
============

[Google Scholar](https://scholar.google.com/citations?user=MeQGlRwAAAAJ&hl=en)

  <ul>{% for post in site.publications %}
    {% include archive-single-cv.html %}
  {% endfor %}</ul>
