+++
title = "Unified, community-developed analysis guidelines and templates for multi-omics data interpretability"
date = 2025-05-15T14:15:00-05:00  # Schedule page publish date.
draft = false

# Talk start and end times.
#   End time can optionally be hidden by prefixing the line with `#`.
time_start = 2025-05-15T14:15:00-05:00

# Authors. Comma separated list, e.g. `["Bob Smith", "David Jones"]`.
authors = ["Nextflow", "Lorena Pantano"]

# Abstract and optional shortened version.
abstract = "Biomedical research generates complex, large-scale data that presents challenges in manipulation, storage, distribution, and analysis, particularly for non-computational biologists. Advancements in sequencing technologies, including single-cell and spatial biology, have transformed biological research approaches. The increasing volume and complexity of sequencing data necessitate automated, scalable infrastructure for reproducible data processing and analysis. Platforms like Nextflow and nf-core have democratized access to primary sequencing analysis through standardized, reproducible bioinformatics pipelines. Specialized bioinformatics cores have become essential in supporting researchers, providing expertise in statistical analysis and interpretation. The Harvard Chan Bioinformatics Core, for example, analyzes numerous projects annually, contributing to publications and offering consultations on experimental design and methodology. While data pre-processing has become more standardized, researchers often struggle with complex study designs and appropriate bioinformatics methods. The bcbio-reports project aims to streamline downstream data analysis by providing report templates for various data types and analyses in a single resource. This infrastructure standardizes downstream multi-omic data analysis, allowing researchers to focus on interpretation and biological understanding. The vision of bcbio emphasizes quantifiability, community development, reproducibility, accessibility, portability, and cutting-edge methodologies. This approach aims to overcome challenges in maintaining complex packages in a rapidly evolving research landscape while ensuring reliable and reproducible results."
abstract_short = ""

# Name of event and optional event URL.
event = "Nextflow Summit 2025"
event_url = "https://summit.nextflow.io/2025/boston"

# Location of event.
# location = "CLS 421 (Cancer Center conference room).4th floor, Center for Life Sciences.3 Blackfan Circle, Boston MA"

# Is this a selected talk? (true/false)
selected = true

# Projects (optional).
#   Associate this talk with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["deep-learning"]` references
#   `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects = []

# Slides (optional).
#   Associate this page with Markdown slides.
#   Simply enter your slide deck's filename without extension.
#   E.g. `slides = "example-slides"` references
#   `content/slides/example-slides.md`.
#   Otherwise, set `slides = ""`.
slides = ""

# Tags (optional).
#   Set `tags = []` for no tags, or use the form `tags = ["A Tag", "Another Tag"]` for one or more tags.
tags = []

# Links (optional).
url_pdf = ""
url_slides = ""
url_video = "https://youtu.be/mi8Wj32oAWI?si=6BEhNqTULjGK2bKW"
url_code = ""

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
[image]
  # Caption (optional)
  caption = ""

  # Focal point (optional)
  # Options: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight
  focal_point = ""
+++
