<!DOCTYPE html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: January 16, 2026 --><html lang="en-us" >


<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  
  
  
    <meta name="generator" content="Hugo Blox Builder 5.9.7" />
  

  
  












  
  










  







  
  
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
  

  
  
  
    
      
      <link rel="preload" as="style" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap">
      <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap" media="print" onload="this.media='all'">
    
  

  
  

  
  

  

  <link rel="stylesheet" href="/css/vendor-bundle.min.26c458e6907dc03073573976b7f4044e.css" media="print" onload="this.media='all'">

  
  
  
    
    

    
    
    
    
      
      
    
    
    

    
    
    

    

    
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      
        
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
  

  
  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.17ed658cdb0bf2fb0768cb75334dcc1c.css" />

  
  
  

  
  
  
  
  
  
  
    
    
    <link rel="stylesheet" href="/css/libs/chroma/github-light.min.css" title="hl-light" media="print" onload="this.media='all'" >
    <link rel="stylesheet" href="/css/libs/chroma/dracula.min.css" title="hl-dark" media="print" onload="this.media='all'" disabled>
  

  
  
























  
  
  










  

<meta name="description" content="In summary: I will show which is the best miRNA mapping tool. I used several optionsÂ for this benchmarking:

miraligner fromÂ SeqBusterÂ suit (I am one of the authors)
bowtie2 and bowtie
novoalignÂ from novocraft suit
GEM
srnabench
microrazer
STAR
miRExpress

I think that these are the most used, and other not used but good to try them. They were clearly developed for other purposes, but as well, they generate the input of many miRNA pipelines. I just wanted to know how well my tool was doing. The first aim to develop miraligner was to get annotated additions of nucleotides at the end of miRNA sequences, something that is very common in mirna biogenesis: isomirsÂ and often they are missed by short read and fast mappers. I have aÂ repositoryÂ for this kind of things, so anybody can reproduce my results, and check if I did something wrong, or comment on it. In this post I just want to know which tool detects more miRNA, for that I did two main steps:" />



<link rel="alternate" hreflang="en-us" href="http://lpantano.github.io/post/legacy/2016-03-24-mirna-annotation-tools-comparison/" />
<link rel="canonical" href="http://lpantano.github.io/post/legacy/2016-03-24-mirna-annotation-tools-comparison/" />



  <link rel="manifest" href="/manifest.webmanifest" />



<link rel="icon" type="image/png" href="/media/icon_hu_6ba9f341fba7588.png" />
<link rel="apple-touch-icon" type="image/png" href="/media/icon_hu_f016f6a28e4542.png" />

<meta name="theme-color" content="#457B9D" />










  
  






<meta property="twitter:card" content="summary" />
<meta property="twitter:image" content="http://lpantano.github.io/media/icon_hu_518ecd45d9711c5b.png" />



  

<meta property="og:type" content="article" />
<meta property="og:site_name" content="My bits" />
<meta property="og:url" content="http://lpantano.github.io/post/legacy/2016-03-24-mirna-annotation-tools-comparison/" />
<meta property="og:title" content="miRNA Annotation Tools Comparison | My bits" />
<meta property="og:description" content="In summary: I will show which is the best miRNA mapping tool. I used several optionsÂ for this benchmarking:

miraligner fromÂ SeqBusterÂ suit (I am one of the authors)
bowtie2 and bowtie
novoalignÂ from novocraft suit
GEM
srnabench
microrazer
STAR
miRExpress

I think that these are the most used, and other not used but good to try them. They were clearly developed for other purposes, but as well, they generate the input of many miRNA pipelines. I just wanted to know how well my tool was doing. The first aim to develop miraligner was to get annotated additions of nucleotides at the end of miRNA sequences, something that is very common in mirna biogenesis: isomirsÂ and often they are missed by short read and fast mappers. I have aÂ repositoryÂ for this kind of things, so anybody can reproduce my results, and check if I did something wrong, or comment on it. In this post I just want to know which tool detects more miRNA, for that I did two main steps:" /><meta property="og:image" content="http://lpantano.github.io/media/icon_hu_518ecd45d9711c5b.png" /><meta property="og:locale" content="en-us" />

  
    <meta
      property="article:published_time"
      content="2016-03-24T19:49:01-04:00"
    />
  
  
    <meta property="article:modified_time" content="2016-03-24T19:49:01-04:00">
  






    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://lpantano.github.io/post/legacy/2016-03-24-mirna-annotation-tools-comparison/"
  },
  "headline": "miRNA Annotation Tools Comparison",
  
  "datePublished": "2016-03-24T19:49:01-04:00",
  "dateModified": "2016-03-24T19:49:01-04:00",
  
  "publisher": {
    "@type": "Organization",
    "name": "My bits",
    "logo": {
      "@type": "ImageObject",
      "url": "http://lpantano.github.io/media/icon_hu_de15e77181fecba7.png"
    }
  },
  "description": "\u003cp\u003eIn summary: I will show which is the best miRNA mapping tool. I used several optionsÂ for this benchmarking:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003emiraligner fromÂ \u003ca href=\"http://github.com/lpantano/seqbuster\" target=\"_blank\" rel=\"noopener\"\u003eSeqBuster\u003c/a\u003eÂ suit (I am one of the authors)\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://bowtie-bio.sourceforge.net/bowtie2/index.shtml\" target=\"_blank\" rel=\"noopener\"\u003ebowtie2\u003c/a\u003e and \u003ca href=\"http://bowtie-bio.sourceforge.net/index.shtml\" target=\"_blank\" rel=\"noopener\"\u003ebowtie\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://www.novocraft.com/main/index.php\" target=\"_blank\" rel=\"noopener\"\u003enovoalign\u003c/a\u003eÂ from novocraft suit\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://algorithms.cnag.cat/wiki/The_GEM_library\" target=\"_blank\" rel=\"noopener\"\u003eGEM\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://arn.ugr.es/srnabench/\" target=\"_blank\" rel=\"noopener\"\u003esrnabench\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://www.seqan.de/projects/MicroRazerS.html.\" target=\"_blank\" rel=\"noopener\"\u003emicrorazer\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/alexdobin/STAR\" target=\"_blank\" rel=\"noopener\"\u003eSTAR\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"http://mirexpress.mbc.nctu.edu.tw/\" target=\"_blank\" rel=\"noopener\"\u003emiRExpress\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eI think that these are the most used, and other not used but good to try them. They were clearly developed for other purposes, but as well, they generate the input of many miRNA pipelines. I just wanted to know how well my tool was doing. The first aim to develop miraligner was to get annotated additions of nucleotides at the end of miRNA sequences, something that is very common in mirna biogenesis: \u003ca href=\"http://en.wikipedia.org/wiki/IsomiR\" target=\"_blank\" rel=\"noopener\"\u003eisomirs\u003c/a\u003eÂ and often they are missed by short read and fast mappers. I have aÂ \u003ca href=\"https://github.com/lpantano/mypubs/tree/master/mirna/mirannotation\" target=\"_blank\" rel=\"noopener\"\u003erepository\u003c/a\u003eÂ for this kind of things, so anybody can reproduce my results, and check if I did something wrong, or comment on it. In this post I just want to know which tool detects more miRNA, for that I did two main steps:\u003c/p\u003e"
}
</script>

  

  




  
  
  

  
  

  


  
  <title>miRNA Annotation Tools Comparison | My bits</title>

  
  
  
  











</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper   " data-wc-page-id="668958435b87a09c9705e59cf5eaabc9" >

  
  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.9efdeed80f3d37db59da84ebeead9552.js"></script>

  


<aside class="search-modal" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#" aria-label="Close"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control"
        aria-label="Search...">
        
      </div>

      
      

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header header--fixed">
  
  
  
  
  












<header>
  <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
    <div class="container-xl">

      
      <div class="d-none d-lg-inline-flex">
        <a class="navbar-brand" href="/">My bits</a>
      </div>
      

      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar-content" aria-controls="navbar-content" aria-expanded="false" aria-label="Toggle navigation">
      <span><i class="fas fa-bars"></i></span>
      </button>
      

      
      <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
        <a class="navbar-brand" href="/">My bits</a>
      </div>
      

      
      
      <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

        
        <ul class="navbar-nav d-md-inline-flex">
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#about"><span>Home</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#projects"><span>Portfolio</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/tutorials/"><span>Tutorials (Coming Soon)</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/experience/"><span>Experience</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link  active" href="/post/"><span>Posts</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/talk/"><span>Talks</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/publication/"><span>Publications</span></a>
          </li>

          
          

        

          
        </ul>
      </div>

      <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

        
        
          
        

        
        
        
        <li class="nav-item">
          <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        
        
        
        <li class="nav-item dropdown theme-dropdown">
          <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
            <i class="fas fa-moon" aria-hidden="true"></i>
          </a>
          <div class="dropdown-menu">
            <a href="#" class="dropdown-item js-set-theme-light">
              <span>Light</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-dark">
              <span>Dark</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-auto">
              <span>Automatic</span>
            </a>
          </div>
        </li>
        

        
        

      </ul>

    </div>
  </nav>
</header>


  </div>

  
  
<div class="newsletter-banner-container" style="background: #ffa640; border-bottom: 2px solid rgba(0,0,0,0.1); position: sticky; top: 0; z-index: 999;">
  <div class="container" style="max-width: 1200px; margin: 0 auto; padding: 0.75rem 1rem;">
    <div style="display: flex; justify-content: center; align-items: center; gap: 1rem; flex-wrap: wrap; text-align: center;">
      <span style="color: #333; font-weight: 500; font-size: 0.95rem; margin: 0;">
        ðŸ“¬ Stay updatedâ€”subscribe to <strong>Decoding Data</strong>
      </span>
      <a class="btn btn-light btn-sm" 
         style="padding: 0.4rem 1rem; font-weight: 600; border-radius: 20px; border: none; box-shadow: 0 2px 4px rgba(0,0,0,0.2); transition: all 0.2s ease;" 
         href="https://decodingdata.substack.com/?utm_campaign=pub&utm_medium=web" 
         target="_blank" 
         rel="noopener"
         onmouseover="this.style.transform='scale(1.05)'; this.style.boxShadow='0 4px 8px rgba(0,0,0,0.3)';"
         onmouseout="this.style.transform='scale(1)'; this.style.boxShadow='0 2px 4px rgba(0,0,0,0.2)';">
        Subscribe Now
      </a>
    </div>
  </div>
</div>


  <div class="page-body">
    
    
    

    <article class="article">

  













  

  
  
  
<div class="article-container pt-3">
  <h1>miRNA Annotation Tools Comparison</h1>

  

  
    


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    Mar 24, 2016
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    3 min read
  </span>
  

  
  
  
  

  
  

</div>

    





  
</div>



  <div class="article-container">

    <div class="article-style">
      <p>In summary: I will show which is the best miRNA mapping tool. I used several optionsÂ for this benchmarking:</p>
<ul>
<li>miraligner fromÂ <a href="http://github.com/lpantano/seqbuster" target="_blank" rel="noopener">SeqBuster</a>Â suit (I am one of the authors)</li>
<li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank" rel="noopener">bowtie2</a> and <a href="http://bowtie-bio.sourceforge.net/index.shtml" target="_blank" rel="noopener">bowtie</a></li>
<li><a href="http://www.novocraft.com/main/index.php" target="_blank" rel="noopener">novoalign</a>Â from novocraft suit</li>
<li><a href="http://algorithms.cnag.cat/wiki/The_GEM_library" target="_blank" rel="noopener">GEM</a></li>
<li><a href="http://arn.ugr.es/srnabench/" target="_blank" rel="noopener">srnabench</a></li>
<li><a href="http://www.seqan.de/projects/MicroRazerS.html." target="_blank" rel="noopener">microrazer</a></li>
<li><a href="https://github.com/alexdobin/STAR" target="_blank" rel="noopener">STAR</a></li>
<li><a href="http://mirexpress.mbc.nctu.edu.tw/" target="_blank" rel="noopener">miRExpress</a></li>
</ul>
<p>I think that these are the most used, and other not used but good to try them. They were clearly developed for other purposes, but as well, they generate the input of many miRNA pipelines. I just wanted to know how well my tool was doing. The first aim to develop miraligner was to get annotated additions of nucleotides at the end of miRNA sequences, something that is very common in mirna biogenesis: <a href="http://en.wikipedia.org/wiki/IsomiR" target="_blank" rel="noopener">isomirs</a>Â and often they are missed by short read and fast mappers. I have aÂ <a href="https://github.com/lpantano/mypubs/tree/master/mirna/mirannotation" target="_blank" rel="noopener">repository</a>Â for this kind of things, so anybody can reproduce my results, and check if I did something wrong, or comment on it. In this post I just want to know which tool detects more miRNA, for that I did two main steps:</p>
<ol>
<li>simulate a bunch of miRNAs (isomirs) with my pythonÂ <a href="https://github.com/lpantano/seqbuster/blob/master/misc/miRNA.simulator.py" target="_blank" rel="noopener">script</a>Â that is part of SeqBuster suit. I generated around 10000 sequences. Normally, one small-RNAseq libraryÂ produces around half million different sequences.</li>
<li>use only miRBase human precursors as reference genome</li>
</ol>
<p>I used default parameter for all, so probably there is a set of parameters that would be better for some tools, but I didn&rsquo;t search for them this time (happy to accept issues to add them). TheÂ <a href="http://rawgit.com/lpantano/mypubs/master/mirna/mirannotation/stats.html" target="_blank" rel="noopener">results</a>Â are:</p>
<ol>
<li><strong>miraligner</strong> and <strong>STAR</strong> map <strong>more sequences</strong>. miraligner loses sequences shorter than 15 nt, normally (miRNA) are around 21, and those sequences map to repeat elements</li>
<li>STAR is the best mapping, but need some parsing to reduce false positive. I think that pipelines should change to this tool.</li>
<li><strong>GEM</strong> has problems with <strong>additions and nt substitutions</strong> in many cases, same as novoaligner (I will look at this in the future)</li>
<li><strong>bowtie2/bowtie is the second</strong> that annotated most (and best)</li>
<li><strong>microrazer</strong> has a problem with <strong>mismatches</strong>, but maybe there is some parameter to trick</li>
<li><strong>miRExpress</strong> with default options only will map perfect matches sequences to precursor, so strongly recommended to allow errors to increase sensibility.</li>
</ol>
<p>I have to say that the advantage of <a href="https://github.com/lpantano/seqbuster/wiki/miraligner" target="_blank" rel="noopener">miraligner</a> is that it gives you the sequence annotated as miRNA or precursor, and gives you the exactly modifications that sequences have if they are compared to miRBase database. And you could feed the results to my <a href="https://github.com/lpantano/seqbuster/wiki/isomiRs" target="_blank" rel="noopener">R package</a> to plot isomers distribution of samples from different groups, and do differential expression analysis with DESeq2, or another tool. I didn&rsquo;t add time consumption because all of them run in a couple of minutes. In my next post I will focus in the the correct annotation of each sequence, and the possible problems with cross-mapping events, when the sequence comes from another regions of the genome but map to miRBase precursor as well [see my previousÂ <a href="http://lorenapantano.wordpress.com/2013/05/27/mirna-annotation-complex-scenarios/" target="_blank" rel="noopener">post</a>Â for more details]. As well I will use STAR with the full genome and see if the mapping continues being the best. In that case I will add a script to SeqBuster to parse the output of STAR for those who can map with STAR (need up to 32G for human genome)</p>

    </div>

    





<div class="article-tags">
  
  <a class="badge badge-light" href="/tag/mirna/">MiRNA</a>
  
</div>



<div class="share-box">
  <ul class="share">
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flpantano.github.io%2Fpost%2Flegacy%2F2016-03-24-mirna-annotation-tools-comparison%2F&amp;text=miRNA&#43;Annotation&#43;Tools&#43;Comparison" target="_blank" rel="noopener" class="share-btn-twitter" aria-label="twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flpantano.github.io%2Fpost%2Flegacy%2F2016-03-24-mirna-annotation-tools-comparison%2F&amp;t=miRNA&#43;Annotation&#43;Tools&#43;Comparison" target="_blank" rel="noopener" class="share-btn-facebook" aria-label="facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
        
      
      <li>
        <a href="mailto:?subject=miRNA%20Annotation%20Tools%20Comparison&amp;body=http%3A%2F%2Flpantano.github.io%2Fpost%2Flegacy%2F2016-03-24-mirna-annotation-tools-comparison%2F" target="_blank" rel="noopener" class="share-btn-email" aria-label="envelope">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flpantano.github.io%2Fpost%2Flegacy%2F2016-03-24-mirna-annotation-tools-comparison%2F&amp;title=miRNA&#43;Annotation&#43;Tools&#43;Comparison" target="_blank" rel="noopener" class="share-btn-linkedin" aria-label="linkedin-in">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="whatsapp://send?text=miRNA&#43;Annotation&#43;Tools&#43;Comparison%20http%3A%2F%2Flpantano.github.io%2Fpost%2Flegacy%2F2016-03-24-mirna-annotation-tools-comparison%2F" target="_blank" rel="noopener" class="share-btn-whatsapp" aria-label="whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=http%3A%2F%2Flpantano.github.io%2Fpost%2Flegacy%2F2016-03-24-mirna-annotation-tools-comparison%2F&amp;title=miRNA&#43;Annotation&#43;Tools&#43;Comparison" target="_blank" rel="noopener" class="share-btn-weibo" aria-label="weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>











  
  



  
  
  
    
  
  
  
  <div class="media author-card content-widget-hr">
    
      
      <a href="http://lpantano.github.io/"><img class="avatar mr-3 avatar-circle" src="/author/avatar_hu_44c7b90fe395a2f7.jpg" alt=""></a>
    

    <div class="media-body">
      <h5 class="card-title"><a href="http://lpantano.github.io/"></a></h5>
      <h6 class="card-subtitle">Director, Bioinformatics Platform</h6>
      <p class="card-text">I build tools, visualizations, and platforms that turn genomic data into actionable insights for drug discovery and target identification.</p>
      <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://www.linkedin.com/in/lpantano/" target="_blank" rel="noopener">
        <i class="fab fa-linkedin"></i>
      </a>
    </li>
  
    
    
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.co.uk/citations?user=MeQGlRwAAAAJ&amp;hl" target="_blank" rel="noopener">
        <i class="ai ai-google-scholar"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/lpantano" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
</ul>

    </div>
  </div>


















  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">

  












  
  
  
  
  













  





  <p class="powered-by">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> â€” the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  


<script src="/js/vendor-bundle.min.6782e3f6a4f06ea4766df038c45ecf40.js"></script>




  

  
  

  













  
  <script id="search-hit-fuse-template" type="text/x-template">
    <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
    </div>
  </script>
  
    <script src="https://cdn.jsdelivr.net/gh/krisk/Fuse@v3.2.1/dist/fuse.min.js" integrity="sha512-o38bmzBGX+hD3JHWUFCDA09btWaqrNmoJ3RXLlrysA7PP01Kgs4UlE4MhelE1v5dJR3+cxlR4qQlotsW7jKsnw==" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/gh/julmot/mark.js@8.11.1/dist/jquery.mark.min.js" integrity="sha512-mhbv5DqBMgrWL+32MmsDOt/OAvqr/cHimk6B8y/bx/xS88MVkYGPiVv2ixKVrkywF2qHplNRUvFsAHUdxZ3Krg==" crossorigin="anonymous"></script>
  












  
  
  
  
  
  
  

















<script id="page-data" type="application/json">{"use_headroom":true}</script>


  <script src="/js/wowchemy-headroom.b95a7e109243f29d04930ae8cb49a756.js" type="module"></script>









  
  


<script src="/en/js/wowchemy.min.69fc52a6dc16aaf1a452cfbb36466dac.js"></script>







  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        
        <pre><code></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>


  <script src="/js/wowchemy-publication.9c0e895144aef5a693008b5c5d450147.js" type="module"></script>


















</body>
</html>
