---
title: Microarray and deep sequencing cross-platform analysis of the mirRNome and isomiR variation in response to epidermal growth factor.
collection: publications
permalink: /publications/2013-BMCgenomics-20
venue: "BMC genomics"
paperurl: www.ncbi.nlm.nih.gov/pubmed/?term=23724959
citation: 'F Llorens,M Hummel,**L Pantano**,X Pastor,A Vivancos,E Castillo,H Mattlin,A Ferrer,M Ingham,M Noguera,R Kofler,J Dohm,R Pluvinet,M Bayes,H Himmelbauer,R del,E Marti,L Sumoy (2013) Microarray and deep sequencing cross-platform analysis of the mirRNome and isomiR variation in response to epidermal growth factor. <i>BMC genomics</i>'
---

F Llorens,M Hummel,**L Pantano**,X Pastor,A Vivancos,E Castillo,H Mattlin,A Ferrer,M Ingham,M Noguera,R Kofler,J Dohm,R Pluvinet,M Bayes,H Himmelbauer,R del,E Marti,L Sumoy
## Abstract
BACKGROUND: Epidermal Growth Factor (EGF) plays an important function in the regulation of cell growth, proliferation, and differentiation by binding to its receptor (EGFR) and providing cancer cells with increased survival responsiveness. Signal transduction carried out by EGF has been extensively studied at both transcriptional and post-transcriptional levels. Little is known about the involvement of microRNAs (miRNAs) in the EGF signaling pathway. miRNAs have emerged as major players in the complex networks of gene regulation, and cancer miRNA expression studies have evidenced a direct involvement of miRNAs in cancer progression.

RESULTS: In this study, we have used an integrative high content analysis approach to identify the specific miRNAs implicated in EGF signaling in HeLa cells as potential mediators of cancer mediated functions. We have used microarray and deep-sequencing technologies in order to obtain a global view of the EGF miRNA transcriptome with a robust experimental cross-validation. By applying a procedure based on Rankprod tests, we have delimited a solid set of EGF-regulated miRNAs. After validating regulated miRNAs by reverse transcription quantitative PCR, we have derived protein networks and biological functions from the predicted targets of the regulated miRNAs to gain insight into the potential role of miRNAs in EGF-treated cells. In addition, we have analyzed sequence heterogeneity due to editing relative to the reference sequence (isomiRs) among regulated miRNAs.

CONCLUSIONS: We propose that the use of global genomic miRNA cross-validation derived from high throughput technologies can be used to generate more reliable datasets inferring more robust networks of co-regulated predicted miRNA target genes.
