<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.4.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Lorena Pantano">

  
  
  
    
  
  <meta name="description" content="I found myself to force to use cellranger. Meanwhile it helps a lot to run from bcl files to single cell counts matrixes, I discovered that is quite difficult to control many options related to optimization.
I have to run more than 200 samples in a short time of period. In my current position at MIT, I joined the OpenMind cluster in the McGovern institute. I was pleased to find a very flexible cluster, but as any other cluster you need to respect other users.">

  
  <link rel="alternate" hreflang="en-us" href="http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/">

  


  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css" crossorigin="anonymous" title="hl-light">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark" disabled>
        
      
    

    

    

  

  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700|Roboto:400,400italic,700|Roboto+Mono&display=swap">
  

  
  
  
  <link rel="stylesheet" href="/css/academic.min.3bc694af15fd1e4ff7262e7dd46f11a8.css">

  

  
  
  

  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/">

  
  
  
  
    
    
  
  <meta property="twitter:card" content="summary">
  
  <meta property="twitter:site" content="@lopantano">
  <meta property="twitter:creator" content="@lopantano">
  
  <meta property="og:site_name" content="My BioBits">
  <meta property="og:url" content="http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/">
  <meta property="og:title" content="How to set up cellranger to make your hpc admin happy | My BioBits">
  <meta property="og:description" content="I found myself to force to use cellranger. Meanwhile it helps a lot to run from bcl files to single cell counts matrixes, I discovered that is quite difficult to control many options related to optimization.
I have to run more than 200 samples in a short time of period. In my current position at MIT, I joined the OpenMind cluster in the McGovern institute. I was pleased to find a very flexible cluster, but as any other cluster you need to respect other users."><meta property="og:image" content="http://lpantano.github.io/img/icon-192.png">
  <meta property="twitter:image" content="http://lpantano.github.io/img/icon-192.png"><meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2019-07-11T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2019-07-11T00:00:00&#43;00:00">
  

  


  





  <title>How to set up cellranger to make your hpc admin happy | My BioBits</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0 compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">My BioBits</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav mr-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>Home</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#featured"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#talks"><span>Talks</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span>Contact</span></a>
        </li>

        
        

      
      </ul>
      <ul class="navbar-nav ml-auto">
      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        
        <li class="nav-item">
          <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
        </li>
        

      </ul>

    </div>
  </div>
</nav>


  <article class="article" itemscope itemtype="http://schema.org/Article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1 itemprop="name">How to set up cellranger to make your hpc admin happy</h1>

  

  
    



<meta content="2019-07-11 00:00:00 &#43;0000 UTC" itemprop="datePublished">
<meta content="2019-07-11 00:00:00 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>Jul 11, 2019</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    5 min read
  </span>
  

  
  

  
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/&amp;text=How%20to%20set%20up%20cellranger%20to%20make%20your%20hpc%20admin%20happy" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/&amp;t=How%20to%20set%20up%20cellranger%20to%20make%20your%20hpc%20admin%20happy" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook-f"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=How%20to%20set%20up%20cellranger%20to%20make%20your%20hpc%20admin%20happy&amp;body=http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/&amp;title=How%20to%20set%20up%20cellranger%20to%20make%20your%20hpc%20admin%20happy" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://web.whatsapp.com/send?text=How%20to%20set%20up%20cellranger%20to%20make%20your%20hpc%20admin%20happy%20http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/&amp;title=How%20to%20set%20up%20cellranger%20to%20make%20your%20hpc%20admin%20happy" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>

    














  
</div>



  <div class="article-container">

    <div class="article-style" itemprop="articleBody">
      

<p>I found myself to force to use <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/cluster-mode" target="_blank">cellranger</a>. Meanwhile it helps a lot to
run from <a href="https://www.illumina.com/informatics/sequencing-data-analysis/sequence-file-formats.html" target="_blank">bcl</a> files to single cell counts matrixes, I discovered that is
quite difficult to control many options related to optimization.</p>

<p>I have to run more than 200 samples in a short time of period. In my current
position at MIT, I joined the OpenMind cluster in the McGovern institute. I was
pleased to find a very flexible cluster, but as any other cluster you need to
respect other users.</p>

<p>My samples are human samples, so I need a lot of memory so <a href="https://github.com/alexdobin/STAR" target="_blank">STAR</a> can run. The
command line is easy to set up:</p>

<p><code>cellranger count --id=SAMPLE --transcriptome=GRCh38_pre_mRNA --fastqs=PATH2FASTQS --sample=SAMPLE</code></p>

<p>Of course, that is in local mode. If you want to set up a job to a cluster,
it would look like this:</p>

<pre><code>#!/bin/bash
#SBATCH -N 1
#SBATCH -c 6
#SBATCH --mem=12000
#SBATCH -t 0-48:00:00
#SBATCH -J cellr
#SBATCH -e job-counts.e
#SBATCH -o job-counts.o
## SBATCH --mail-type=END,FAIL # this line is commented
## SBATCH --mail-user=user@mit.edu  # this line is commented

cellranger count --id=SAMPLE --transcriptome=GRCh38_pre_mRNA \
          --fastqs=PATH2FASTQS --sample=SAMPLE \
          --localcores=6 --localmem=120

</code></pre>

<p>The main problem with that is you are using 120G during all the process that could
go to 1-2 days.</p>

<p>I discovered that you can setup the <code>cluster</code> mode to make <code>cellranger</code> to send
jobs to the cluster. However, <code>slurm</code> is not a option to the documentation, BUT
you can specify a template file:</p>

<pre><code>#!/bin/bash
#SBATCH -J __MRO_JOB_NAME__
#SBATCH -N 1
#SBATCH -c __MRO_THREADS__
#SBATCH --mem=__MRO_MEM_GB__G
#SBATCH -o __MRO_STDOUT__
#SBATCH -e __MRO_STDERR__
#SBATCH -t 0-48:00:00
#SBATCH --qos=&quot;normal&quot;

 __MRO_CMD__
</code></pre>

<p>The command would look like this:</p>

<pre><code>cellranger count --id=SAMPLE --transcriptome=GRCh38_pre_mRNA \
          --fastqs=PATH2FASTQS --sample=SAMPLE \
          --jobmode=slurm.template \
          --maxjobs=3 --jobinterval=1000

</code></pre>

<p>I was feeling victory, ha ha ha, BUT I saw that the <code>ALIGN</code> step was requesting
220G of memory&hellip;. WHAT!!!!!????? or better <a href="https://www.destroyallsoftware.com/talks/wat" target="_blank">WATTT????!!!!!</a>.</p>

<p><strong>note</strong>: this is set up during genome generation with the parameters:</p>

<pre><code>
    --memgb=&lt;num&gt;       Maximum memory (GB) used when aligning reads with STAR.
                          Defaults to 16.
</code></pre>

<p><strong>note</strong>: default is 16G but if I use that then my jobs are cancelled because using more memory than requested, as I expected considering STAR needs more than that for human genome.</p>

<p>I guess I didn&rsquo;t look at the first place I should have looked, but it took me
some time to discovered that the memory is coming from a file in the reference
genome folder:</p>

<pre><code>cat GRCh38_pre_mRNA/reference.json
{
    &quot;fasta_hash&quot;: &quot;954a9c3916ef1544c9358445440b9683fc061c71&quot;,
    &quot;genomes&quot;: [
        &quot;GRCh38_pre_mRNA&quot;
    ],
    &quot;gtf_hash&quot;: &quot;12aae438159b35282adaf643cca4dc5887b0448a&quot;,
    &quot;input_fasta_files&quot;: [
        &quot;Homo_sapiens.GRCh38.dna.toplevel.fa&quot;
    ],
    &quot;input_gtf_files&quot;: [
        &quot;GRC38_preRNA.gtf&quot;
    ],
    &quot;mem_gb&quot;: 220,
    &quot;mkref_version&quot;: &quot;3.0.2&quot;,
    &quot;threads&quot;: 24,
    &quot;version&quot;: null
}
</code></pre>

<p>I realized that when I found, after 100 <code>ls/cat/less/grep</code>, this file with these
lines of code:</p>

<pre><code>cellranger-3.0.2/cellranger-cs/3.0.2/lib/python/cellranger/utils.py
def _load_reference_metadata_file(reference_path):
    reference_metadata_file = os.path.join(reference_path, cr_constants.REFERENCE_METADATA_FILE)
    with open(reference_metadata_file, 'r') as f:
        return json.load(f)
</code></pre>

<p>So, easy peasy, I changed that to 110. I relaunched the job I was using to test,
and found that it was still setting up 220G&hellip;WHY!?!!? well, it seems if you relaunch
in a step where the alignment has already begun, there is a cache for the resources.
So after more time trying to figure out this and asking myself, why I decided my
career path to be bioinformatics, I tried a fresh start and&hellip;IT WORKED.</p>

<blockquote>
<p>How happy one can be to see a job requesting the expected resources.</p>

<p>Incredible happy!</p>
</blockquote>

<p>I went a step further, and looked how much memory actually it was using during the
alignment step, and I saw that actually only needs 64G, so I updated to 72G, to
gain more karma from the HPC world.</p>

<p>There is one more thing, that will set jobs of 72G and 4 cores, if you want to
increase the number of cores, then use this option <code>--mempercore 6</code> or whatever number
that you want: <code>mempercore = 72G/final_cores</code>.</p>

<h2 id="workflow">Workflow</h2>

<p>Finally, I felt that I was doing my best to behave and avoid making angry other
people using the hpc. The final setup looks like this:</p>

<ul>
<li><p>meta-script to send all the samples to a master job: <code>runner.sh</code>. It accepts
  a file where first column is <code>sample name</code> and second column is <code>fastq_path</code></p>

<pre><code>set -o pipefail  # trace ERR through pipes
set -o errtrace  # trace ERR through 'time command' and other functions
set -o nounset   ## set -u : exit the script if you try to use an uninitialised variable
set -o errexit   ## set -e : exit the script if any statement returns a non-true return value
# set -v # you can uncomment all these lines to have a better debugging
# set -x
# export PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'

while IFS=&quot; &quot; read -r sample paths

do

RUN=0
if [[ ! -e $sample/filtered_feature_bc_matrix.h5 ]]; then
RUN=1 # run if the final output is not there yet
fi

if [[  -e  $sample/_lock ]]; then
echo $sample is locked, please:
echo rm $sample/_lock
RUN=0 # but don't do anything if is locked
fi

if [[ $RUN == 1 ]]; then
echo sbatch -J $sample -o logs/$sample.o -e logs/$sample.e scripts/om-counts-cluster.slurm $sample $paths
sbatch -J $sample -o logs/$sample.o -e logs/$sample.e scripts/om-counts-cluster.slurm $sample $paths
fi

done &lt; $1
</code></pre></li>

<li><p>SBATCH script with cellranger command: <code>counter.slurm</code></p>

<pre><code>#!/bin/bash
#SBATCH -N 1
#SBATCH -c 1
#SBATCH --mem=6000
#SBATCH -t 0-48:00:00
# SBATCH -J &quot;cellr-$1&quot;
# SBATCH -e job-counts-$1.e
# SBATCH -o job-counts-$1.o
## SBATCH --mail-type=END,FAIL # this line is commented
## SBATCH --mail-user=user@mit.edu  # this line is commented

SCRIPTPATH=&quot;PATHTOSCRIPTS&quot;
cellranger count --id=$1 --transcriptome=GRCh38_pre_mRNA --fastqs=$2 --sample=$1 --jobmode=$SCRIPTPATH/slurm.template --maxjobs=3 --jobinterval=1000

if [ $? -eq 0 ]; then
echo OK
mv $1 results/.
else
echo $1 FAIL
fi
</code></pre></li>

<li><p>template script for <code>cellranger</code> as shown above</p></li>
</ul>

<h3 id="other-tips">Other tips</h3>

<p>I used <code>mxjobs=3</code> to avoid a lot of jobs. I am testing how things go, this could be spanned to bigger numbers.</p>

<p>I used <code>jobinterval=1000</code> to avoid sending jobs at the same time, just in case, but I don&rsquo;t have any proof to support this.</p>

<blockquote>
<p>&ldquo;happiness in bioinformatics:</p>
</blockquote>

    </div>

    

<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/hpc/">hpc</a>
  
  <a class="badge badge-light" href="/tags/singlecell/">singlecell</a>
  
</div>



    
      








  





  
  
  
    
  
  
  <div class="media author-card" itemscope itemtype="http://schema.org/Person">
    
      
      <img class="portrait mr-3" src="/authors/admin/avatar_hua735c4bd12b5368408d6d400f29b21ec_801139_250x250_fill_q90_lanczos_center.jpg" itemprop="image" alt="Avatar">
    

    <div class="media-body">
      <h5 class="card-title" itemprop="name"><a href="http://lpantano.github.io/">Lorena Pantano</a></h5>
      <h6 class="card-subtitle">Senior Computational Biologist</h6>
      <p class="card-text" itemprop="description">My research interests include distributed robotics, mobile computing and programmable matter.</p>
      <ul class="network-icon" aria-hidden="true">
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="/#contact" >
              <i class="fas fa-envelope"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://twitter.com/lopantano" target="_blank" rel="noopener">
              <i class="fab fa-twitter"></i>
            </a>
          </li>
        
          
          
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://scholar.google.co.uk/citations?user=MeQGlRwAAAAJ&amp;hl" target="_blank" rel="noopener">
              <i class="ai ai-google-scholar"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://github.com/lpantano" target="_blank" rel="noopener">
              <i class="fab fa-github"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>



      
      
    

    

    


  </div>
</article>

      

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js" integrity="sha256-aYTdUrn6Ow1DDgh5JTc3aDGnnju48y/1c8s1dgkYPQ8=" crossorigin="anonymous"></script>
        
      

      
      
    

    
    

    
    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.16bbb3750feb7244c9bc409a5a4fe678.js"></script>

    






  
  <div class="container">
    <footer class="site-footer">
  

  <p class="powered-by">
    

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
