<!DOCTYPE html>
<!-- This site was created with Hugo Blox. https://hugoblox.com -->
<!-- Last Published: January 18, 2026 --><html lang="en-us" >


<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  
  
  
    <meta name="generator" content="Hugo Blox Builder 5.9.7" />
  

  
  












  
  










  







  
  
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
  

  
  
  
    
      
      <link rel="preload" as="style" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap">
      <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap" media="print" onload="this.media='all'">
    
  

  
  

  
  

  

  <link rel="stylesheet" href="/css/vendor-bundle.min.26c458e6907dc03073573976b7f4044e.css" media="print" onload="this.media='all'">

  
  
  
    
    

    
    
    
    
      
      
    
    
    

    
    
    

    

    
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      
        
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
  

  
  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.17ed658cdb0bf2fb0768cb75334dcc1c.css" />

  
  
  

  
  
  
  
  
  
  
    
    
    <link rel="stylesheet" href="/css/libs/chroma/github-light.min.css" title="hl-light" media="print" onload="this.media='all'" >
    <link rel="stylesheet" href="/css/libs/chroma/dracula.min.css" title="hl-dark" media="print" onload="this.media='all'" disabled>
  

  
  
























  
  
  










  

<meta name="description" content="I found myself to force to use cellranger. Meanwhile it helps a lot to
run from bcl files to single cell counts matrixes, I discovered that is
quite difficult to control many options related to optimization." />



<link rel="alternate" hreflang="en-us" href="http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/" />
<link rel="canonical" href="http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/" />



  <link rel="manifest" href="/manifest.webmanifest" />



<link rel="icon" type="image/png" href="/media/icon_hu_6ba9f341fba7588.png" />
<link rel="apple-touch-icon" type="image/png" href="/media/icon_hu_f016f6a28e4542.png" />

<meta name="theme-color" content="#457B9D" />










  
  






<meta property="twitter:card" content="summary" />
<meta property="twitter:image" content="http://lpantano.github.io/media/icon_hu_518ecd45d9711c5b.png" />



  

<meta property="og:type" content="article" />
<meta property="og:site_name" content="My bits" />
<meta property="og:url" content="http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/" />
<meta property="og:title" content="How to set up cellranger to make your hpc admin happy | My bits" />
<meta property="og:description" content="I found myself to force to use cellranger. Meanwhile it helps a lot to
run from bcl files to single cell counts matrixes, I discovered that is
quite difficult to control many options related to optimization." /><meta property="og:image" content="http://lpantano.github.io/media/icon_hu_518ecd45d9711c5b.png" /><meta property="og:locale" content="en-us" />

  
    <meta
      property="article:published_time"
      content="2019-07-11T00:00:00&#43;00:00"
    />
  
  
    <meta property="article:modified_time" content="2019-07-11T00:00:00&#43;00:00">
  






    






  




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://lpantano.github.io/post/2019/2019-07-12-cellranger-efficiency-in-hpc-copy/"
  },
  "headline": "How to set up cellranger to make your hpc admin happy",
  
  "datePublished": "2019-07-11T00:00:00Z",
  "dateModified": "2019-07-11T00:00:00Z",
  
  "publisher": {
    "@type": "Organization",
    "name": "My bits",
    "logo": {
      "@type": "ImageObject",
      "url": "http://lpantano.github.io/media/icon_hu_de15e77181fecba7.png"
    }
  },
  "description": "\u003cp\u003eI found myself to force to use \u003ca href=\"https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/cluster-mode\" target=\"_blank\" rel=\"noopener\"\u003ecellranger\u003c/a\u003e. Meanwhile it helps a lot to\nrun from \u003ca href=\"https://www.illumina.com/informatics/sequencing-data-analysis/sequence-file-formats.html\" target=\"_blank\" rel=\"noopener\"\u003ebcl\u003c/a\u003e files to single cell counts matrixes, I discovered that is\nquite difficult to control many options related to optimization.\u003c/p\u003e"
}
</script>

  

  




  
  
  

  
  

  


  
  <title>How to set up cellranger to make your hpc admin happy | My bits</title>

  
  
  
  











</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper   " data-wc-page-id="cb81eb0ae3045bf6d204eb18a2449267" >

  
  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.9efdeed80f3d37db59da84ebeead9552.js"></script>

  


<aside class="search-modal" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#" aria-label="Close"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control"
        aria-label="Search...">
        
      </div>

      
      

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header header--fixed">
  
  
  
  
  












<header>
  <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
    <div class="container-xl">

      
      <div class="d-none d-lg-inline-flex">
        <a class="navbar-brand" href="/">My bits</a>
      </div>
      

      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar-content" aria-controls="navbar-content" aria-expanded="false" aria-label="Toggle navigation">
      <span><i class="fas fa-bars"></i></span>
      </button>
      

      
      <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
        <a class="navbar-brand" href="/">My bits</a>
      </div>
      

      
      
      <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

        
        <ul class="navbar-nav d-md-inline-flex">
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#about"><span>Home</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
              
              
              
                
              
              
            
          

          <li class="nav-item">
            <a class="nav-link " href="/#projects"><span>Portfolio</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/tutorials/"><span>Tutorials (Coming Soon)</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/experience/"><span>Experience</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link  active" href="/post/"><span>Posts</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/talk/"><span>Talks</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/publication/"><span>Publications</span></a>
          </li>

          
          

        

          
        </ul>
      </div>

      <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

        
        
          
        

        
        
        
        <li class="nav-item">
          <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        
        
        
        <li class="nav-item dropdown theme-dropdown">
          <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
            <i class="fas fa-moon" aria-hidden="true"></i>
          </a>
          <div class="dropdown-menu">
            <a href="#" class="dropdown-item js-set-theme-light">
              <span>Light</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-dark">
              <span>Dark</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-auto">
              <span>Automatic</span>
            </a>
          </div>
        </li>
        

        
        

      </ul>

    </div>
  </nav>
</header>


  </div>

  
  
<div class="newsletter-banner-container" style="background: #ffa640; border-bottom: 2px solid rgba(0,0,0,0.1); position: sticky; top: 0; z-index: 999;">
  <div class="container" style="max-width: 1200px; margin: 0 auto; padding: 0.75rem 1rem;">
    <div style="display: flex; justify-content: center; align-items: center; gap: 1rem; flex-wrap: wrap; text-align: center;">
      <span style="color: #333; font-weight: 500; font-size: 0.95rem; margin: 0;">
        ðŸ“¬ Stay updatedâ€”subscribe to <strong>Decoding Data</strong>
      </span>
      <a class="btn btn-light btn-sm" 
         style="padding: 0.4rem 1rem; font-weight: 600; border-radius: 20px; border: none; box-shadow: 0 2px 4px rgba(0,0,0,0.2); transition: all 0.2s ease;" 
         href="https://decodingdata.substack.com/?utm_campaign=pub&utm_medium=web" 
         target="_blank" 
         rel="noopener"
         onmouseover="this.style.transform='scale(1.05)'; this.style.boxShadow='0 4px 8px rgba(0,0,0,0.3)';"
         onmouseout="this.style.transform='scale(1)'; this.style.boxShadow='0 2px 4px rgba(0,0,0,0.2)';">
        Subscribe Now
      </a>
    </div>
  </div>
</div>


  <div class="page-body">
    
    
    

    <article class="article">

  













  

  
  
  
<div class="article-container pt-3">
  <h1>How to set up cellranger to make your hpc admin happy</h1>

  

  
    


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    Jul 11, 2019
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    5 min read
  </span>
  

  
  
  
  

  
  

</div>

    





  
</div>



  <div class="article-container">

    <div class="article-style">
      <p>I found myself to force to use <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/cluster-mode" target="_blank" rel="noopener">cellranger</a>. Meanwhile it helps a lot to
run from <a href="https://www.illumina.com/informatics/sequencing-data-analysis/sequence-file-formats.html" target="_blank" rel="noopener">bcl</a> files to single cell counts matrixes, I discovered that is
quite difficult to control many options related to optimization.</p>
<p>I have to run more than 200 samples in a short time of period. In my current
position at MIT, I joined the OpenMind cluster in the McGovern institute. I was
pleased to find a very flexible cluster, but as any other cluster you need to
respect other users.</p>
<p>My samples are human samples, so I need a lot of memory so <a href="https://github.com/alexdobin/STAR" target="_blank" rel="noopener">STAR</a> can run. The
command line is easy to set up:</p>
<p><code>cellranger count --id=SAMPLE --transcriptome=GRCh38_pre_mRNA --fastqs=PATH2FASTQS --sample=SAMPLE</code></p>
<p>Of course, that is in local mode. If you want to set up a job to a cluster,
it would look like this:</p>
<pre><code>#!/bin/bash
#SBATCH -N 1
#SBATCH -c 6
#SBATCH --mem=12000
#SBATCH -t 0-48:00:00
#SBATCH -J cellr
#SBATCH -e job-counts.e
#SBATCH -o job-counts.o
## SBATCH --mail-type=END,FAIL # this line is commented
## SBATCH --mail-user=user@mit.edu  # this line is commented

cellranger count --id=SAMPLE --transcriptome=GRCh38_pre_mRNA \
          --fastqs=PATH2FASTQS --sample=SAMPLE \
          --localcores=6 --localmem=120

</code></pre>
<p>The main problem with that is you are using 120G during all the process that could
go to 1-2 days.</p>
<p>I discovered that you can setup the <code> cluster</code> mode to make <code>cellranger</code> to send
jobs to the cluster. However, <code>slurm</code> is not a option to the documentation, BUT
you can specify a template file:</p>
<pre><code>#!/bin/bash
#SBATCH -J __MRO_JOB_NAME__
#SBATCH -N 1
#SBATCH -c __MRO_THREADS__
#SBATCH --mem=__MRO_MEM_GB__G
#SBATCH -o __MRO_STDOUT__
#SBATCH -e __MRO_STDERR__
#SBATCH -t 0-48:00:00
#SBATCH --qos=&quot;normal&quot;

 __MRO_CMD__
</code></pre>
<p>The command would look like this:</p>
<pre><code>cellranger count --id=SAMPLE --transcriptome=GRCh38_pre_mRNA \
          --fastqs=PATH2FASTQS --sample=SAMPLE \
          --jobmode=slurm.template \
          --maxjobs=3 --jobinterval=1000

</code></pre>
<p>I was feeling victory, ha ha ha, BUT I saw that the <code>ALIGN</code> step was requesting
220G of memory&hellip;. WHAT!!!!!????? or better <a href="https://www.destroyallsoftware.com/talks/wat" target="_blank" rel="noopener">WATTT????!!!!!</a>.</p>
<p><strong>note</strong>: this is set up during genome generation with the parameters:</p>
<pre><code>
    --memgb=&lt;num&gt;       Maximum memory (GB) used when aligning reads with STAR.
                          Defaults to 16.
</code></pre>
<p><strong>note</strong>: default is 16G but if I use that then my jobs are cancelled because using more memory than requested, as I expected considering STAR needs more than that for human genome.</p>
<p>I guess I didn&rsquo;t look at the first place I should have looked, but it took me
some time to discovered that the memory is coming from a file in the reference
genome folder:</p>
<pre><code>cat GRCh38_pre_mRNA/reference.json
{
    &quot;fasta_hash&quot;: &quot;954a9c3916ef1544c9358445440b9683fc061c71&quot;,
    &quot;genomes&quot;: [
        &quot;GRCh38_pre_mRNA&quot;
    ],
    &quot;gtf_hash&quot;: &quot;12aae438159b35282adaf643cca4dc5887b0448a&quot;,
    &quot;input_fasta_files&quot;: [
        &quot;Homo_sapiens.GRCh38.dna.toplevel.fa&quot;
    ],
    &quot;input_gtf_files&quot;: [
        &quot;GRC38_preRNA.gtf&quot;
    ],
    &quot;mem_gb&quot;: 220,
    &quot;mkref_version&quot;: &quot;3.0.2&quot;,
    &quot;threads&quot;: 24,
    &quot;version&quot;: null
}
</code></pre>
<p>I realized that when I found, after 100 <code>ls/cat/less/grep</code>, this file with these
lines of code:</p>
<pre><code>cellranger-3.0.2/cellranger-cs/3.0.2/lib/python/cellranger/utils.py
def _load_reference_metadata_file(reference_path):
    reference_metadata_file = os.path.join(reference_path, cr_constants.REFERENCE_METADATA_FILE)
    with open(reference_metadata_file, 'r') as f:
        return json.load(f)
</code></pre>
<p>So, easy peasy, I changed that to 110. I relaunched the job I was using to test,
and found that it was still setting up 220G&hellip;WHY!?!!? well, it seems if you relaunch
in a step where the alignment has already begun, there is a cache for the resources.
So after more time trying to figure out this and asking myself, why I decided my
career path to be bioinformatics, I tried a fresh start and&hellip;IT WORKED.</p>
<blockquote>
<p>How happy one can be to see a job requesting the expected resources.</p>
</blockquote>
<blockquote>
<p>Incredible happy!</p>
</blockquote>
<p>I went a step further, and looked how much memory actually it was using during the
alignment step, and I saw that actually only needs 64G, so I updated to 72G, to
gain more karma from the HPC world.</p>
<p>There is one more thing, that will set jobs of 72G and 4 cores, if you want to
increase the number of cores, then use this option <code>--mempercore 6</code> or whatever number
that you want: <code>mempercore = 72G/final_cores</code>.</p>
<h2 id="workflow">Workflow</h2>
<p>Finally, I felt that I was doing my best to behave and avoid making angry other
people using the hpc. The final setup looks like this:</p>
<ul>
<li>meta-script to send all the samples to a master job: <code>runner.sh</code>. It accepts
a file where first column is <code>sample name</code> and second column is <code>fastq_path</code></li>
</ul>
<pre><code>set -o pipefail  # trace ERR through pipes
set -o errtrace  # trace ERR through 'time command' and other functions
set -o nounset   ## set -u : exit the script if you try to use an uninitialised variable
set -o errexit   ## set -e : exit the script if any statement returns a non-true return value
# set -v # you can uncomment all these lines to have a better debugging
# set -x
# export PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'

while IFS=&quot; &quot; read -r sample paths

do

RUN=0
if [[ ! -e $sample/filtered_feature_bc_matrix.h5 ]]; then
    RUN=1 # run if the final output is not there yet
fi

if [[  -e  $sample/_lock ]]; then
    echo $sample is locked, please:
    echo rm $sample/_lock
    RUN=0 # but don't do anything if is locked
fi

if [[ $RUN == 1 ]]; then
    echo sbatch -J $sample -o logs/$sample.o -e logs/$sample.e scripts/om-counts-cluster.slurm $sample $paths
    sbatch -J $sample -o logs/$sample.o -e logs/$sample.e scripts/om-counts-cluster.slurm $sample $paths
fi

done &lt; $1
</code></pre>
<ul>
<li>SBATCH script with cellranger command: <code>counter.slurm</code></li>
</ul>
<pre><code>#!/bin/bash
#SBATCH -N 1
#SBATCH -c 1
#SBATCH --mem=6000
#SBATCH -t 0-48:00:00
# SBATCH -J &quot;cellr-$1&quot;
# SBATCH -e job-counts-$1.e
# SBATCH -o job-counts-$1.o
## SBATCH --mail-type=END,FAIL # this line is commented
## SBATCH --mail-user=user@mit.edu  # this line is commented

SCRIPTPATH=&quot;PATHTOSCRIPTS&quot;
cellranger count --id=$1 --transcriptome=GRCh38_pre_mRNA --fastqs=$2 --sample=$1 --jobmode=$SCRIPTPATH/slurm.template --maxjobs=3 --jobinterval=1000

if [ $? -eq 0 ]; then
    echo OK
    mv $1 results/.
else
    echo $1 FAIL
fi
</code></pre>
<ul>
<li>template script for <code>cellranger</code> as shown above</li>
</ul>
<h3 id="other-tips">Other tips</h3>
<p>I used <code>mxjobs=3</code> to avoid a lot of jobs. I am testing how things go, this could be spanned to bigger numbers.</p>
<p>I used <code>jobinterval=1000</code> to avoid sending jobs at the same time, just in case, but I don&rsquo;t have any proof to support this.</p>
<blockquote>
<p>&ldquo;happiness in bioinformatics:</p>
</blockquote>

    </div>

    





<div class="article-tags">
  
  <a class="badge badge-light" href="/tag/hpc/">Hpc</a>
  
  <a class="badge badge-light" href="/tag/singlecell/">Singlecell</a>
  
</div>



<div class="share-box">
  <ul class="share">
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flpantano.github.io%2Fpost%2F2019%2F2019-07-12-cellranger-efficiency-in-hpc-copy%2F&amp;text=How&#43;to&#43;set&#43;up&#43;cellranger&#43;to&#43;make&#43;your&#43;hpc&#43;admin&#43;happy" target="_blank" rel="noopener" class="share-btn-twitter" aria-label="twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=http%3A%2F%2Flpantano.github.io%2Fpost%2F2019%2F2019-07-12-cellranger-efficiency-in-hpc-copy%2F&amp;t=How&#43;to&#43;set&#43;up&#43;cellranger&#43;to&#43;make&#43;your&#43;hpc&#43;admin&#43;happy" target="_blank" rel="noopener" class="share-btn-facebook" aria-label="facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
        
      
      <li>
        <a href="mailto:?subject=How%20to%20set%20up%20cellranger%20to%20make%20your%20hpc%20admin%20happy&amp;body=http%3A%2F%2Flpantano.github.io%2Fpost%2F2019%2F2019-07-12-cellranger-efficiency-in-hpc-copy%2F" target="_blank" rel="noopener" class="share-btn-email" aria-label="envelope">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=http%3A%2F%2Flpantano.github.io%2Fpost%2F2019%2F2019-07-12-cellranger-efficiency-in-hpc-copy%2F&amp;title=How&#43;to&#43;set&#43;up&#43;cellranger&#43;to&#43;make&#43;your&#43;hpc&#43;admin&#43;happy" target="_blank" rel="noopener" class="share-btn-linkedin" aria-label="linkedin-in">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="whatsapp://send?text=How&#43;to&#43;set&#43;up&#43;cellranger&#43;to&#43;make&#43;your&#43;hpc&#43;admin&#43;happy%20http%3A%2F%2Flpantano.github.io%2Fpost%2F2019%2F2019-07-12-cellranger-efficiency-in-hpc-copy%2F" target="_blank" rel="noopener" class="share-btn-whatsapp" aria-label="whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=http%3A%2F%2Flpantano.github.io%2Fpost%2F2019%2F2019-07-12-cellranger-efficiency-in-hpc-copy%2F&amp;title=How&#43;to&#43;set&#43;up&#43;cellranger&#43;to&#43;make&#43;your&#43;hpc&#43;admin&#43;happy" target="_blank" rel="noopener" class="share-btn-weibo" aria-label="weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>











  
  



  
  
  
    
  
  
  
  <div class="media author-card content-widget-hr">
    
      
      <a href="http://lpantano.github.io/"><img class="avatar mr-3 avatar-circle" src="/author/avatar_hu_44c7b90fe395a2f7.jpg" alt=""></a>
    

    <div class="media-body">
      <h5 class="card-title"><a href="http://lpantano.github.io/"></a></h5>
      <h6 class="card-subtitle">Director, Bioinformatics Platform</h6>
      <p class="card-text">I build tools, visualizations, and platforms that turn genomic data into actionable insights for drug discovery and target identification.</p>
      <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://www.linkedin.com/in/lpantano/" target="_blank" rel="noopener">
        <i class="fab fa-linkedin"></i>
      </a>
    </li>
  
    
    
    
    
    
    
    
      
    
    <li>
      <a href="https://scholar.google.co.uk/citations?user=MeQGlRwAAAAJ&amp;hl" target="_blank" rel="noopener">
        <i class="ai ai-google-scholar"></i>
      </a>
    </li>
  
    
    
    
      
    
    
    
    
    
      
    
    <li>
      <a href="https://github.com/lpantano" target="_blank" rel="noopener">
        <i class="fab fa-github"></i>
      </a>
    </li>
  
</ul>

    </div>
  </div>


















  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">

  












  
  
  
  
  













  





  <p class="powered-by">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://hugoblox.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Hugo Blox Builder</a> â€” the free, <a href="https://github.com/HugoBlox/hugo-blox-builder" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  


<script src="/js/vendor-bundle.min.6782e3f6a4f06ea4766df038c45ecf40.js"></script>




  

  
  

  













  
  <script id="search-hit-fuse-template" type="text/x-template">
    <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
    </div>
  </script>
  
    <script src="https://cdn.jsdelivr.net/gh/krisk/Fuse@v3.2.1/dist/fuse.min.js" integrity="sha512-o38bmzBGX+hD3JHWUFCDA09btWaqrNmoJ3RXLlrysA7PP01Kgs4UlE4MhelE1v5dJR3+cxlR4qQlotsW7jKsnw==" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/gh/julmot/mark.js@8.11.1/dist/jquery.mark.min.js" integrity="sha512-mhbv5DqBMgrWL+32MmsDOt/OAvqr/cHimk6B8y/bx/xS88MVkYGPiVv2ixKVrkywF2qHplNRUvFsAHUdxZ3Krg==" crossorigin="anonymous"></script>
  












  
  
  
  
  
  
  

















<script id="page-data" type="application/json">{"use_headroom":true}</script>


  <script src="/js/wowchemy-headroom.b95a7e109243f29d04930ae8cb49a756.js" type="module"></script>









  
  


<script src="/en/js/wowchemy.min.69fc52a6dc16aaf1a452cfbb36466dac.js"></script>







  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        
        <pre><code></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>


  <script src="/js/wowchemy-publication.9c0e895144aef5a693008b5c5d450147.js" type="module"></script>


















</body>
</html>
